dvc/cache/local.py

```diff
@@ -24,6 +24,14 @@ def __init__(self, tree):
         super().__init__(tree)
         self.cache_dir = tree.config.get("url")
 
+        shared = tree.config.get("shared")
+        if shared:
+            self._file_mode = 0o664
+            self._dir_mode = 0o2775
```

Comment:
Same as git, using S_ISGID sticky bit, this is a more proper way to handle shared cache scenario.